src/GameLibs/Snake/System/AppleSystem.cpp

```diff
@@ -25,23 +28,42 @@ Snake::System::AppleSystem::AppleSystem()
 // TODO check the pos of snake for the apple because we don't want
 // the apple to spawn on the snake
 
-void modifyApplePos(Arcade::ECS::IEntityManager &currentEntityManager, std::vector<Arcade::Vector3f> positions)
+void modifyApplePos(Arcade::ECS::IEventManager &eventManager, Arcade::ECS::IEntityManager &currentEntityManager, std::vector<Arcade::Vector3f> positions)
 {
     std::shared_ptr<Arcade::ECS::IEntity> apple = currentEntityManager.getEntitiesById("Apple");
     Arcade::ECS::IComponent &appleIComp = apple->getComponents("apple");
+    auto snakeEntities = currentEntityManager.getEntitiesByComponentType(Arcade::ECS::CompType::MOVEABLE);
+    std::vector<Arcade::Vector3f> tempPositions = positions;
+    std::size_t nbrOfPos = 20;
+    std::size_t randNumber = 1 + (std::rand() % nbrOfPos);
+    Arcade::Vector3f randPosition = positions[randNumber];
 
+    for (auto it = snakeEntities->begin(); it != snakeEntities->end(); it++) {
+        auto entity = *it;
+        auto components = *(entity->getComponents(Arcade::ECS::CompType::SPRITE).begin());
+        Arcade::Graph::Sprite &moveableSprite = static_cast<Arcade::Graph::Sprite &>(*components);
+
+        if (moveableSprite.pos.x == randPosition.x && moveableSprite.pos.y == randPosition.y) {
+            nbrOfPos -= 1;
+            randPosition = positions[1 + (std::rand() % nbrOfPos)];
+            it = snakeEntities->begin();
```

Comment:
modify it ? in for auto : .. ?

Reply:
Car comme la position est pas bonne il faut que je reparcours toutes les entités du snake pour voir si la nouvelle position générer est correct

Reply:
ahhh oki